src/woocommerce/WC_Gateway_Economic_Invoice.php

```diff
@@ -0,0 +1,263 @@
+                'description' => __('Vælg hvornår fakturaen skal oprettes', 'mt-wc-economic'),
+                'default' => 'creation',
+                'desc_tip' => true,
+            ],
+
+            'economic_vat_zone' => [
+                'title' => __('Economic momszone', 'mt-wc-economic'),
+                'type' => 'select',
+                'options' => $this->getVatZones(),
+                'description' => __('Vælg den ønskede momszone i economic', 'mt-wc-economic'),
+                'default' => null,
+                'desc_tip' => true,
+            ],
+
+            'economic_customer_group' => [
+                'title' => __('Economic Kundegruppe', 'mt-wc-economic'),
+                'type' => 'select',
+                'options' => $this->getCustomerGroup(),
+                'description' => __('Vælg den ønskede kundegruppe i economic', 'mt-wc-economic'),
+                'default' => null,
+                'desc_tip' => true,
+            ],
+
+            'economic_ean' => [
+                'title' => __('Tilføj EAN', 'mt-wc-economic'),
+                'label' => __('Tilføj EAN', 'mt-wc-economic'),
+                'type' => 'checkbox',
+                'description' => __('Skal der være mulighed for at tilføje EAN?', 'mt-wc-economic'),
+                'default' => 'no',
+                'desc_tip' => true,
+            ],
+
+            'economic_shipping_product' => [
+                'title' => __('Economic forsendelses produkt ID', 'mt-wc-economic'),
+                'type' => 'number',
+                'description' => __('Indtast produkt ID tilknyttet fragt i economic', 'mt-wc-economic'),
+                'default' => '',
+                'desc_tip' => true,
+            ],
+        ];
+
+    }
+
+    public function process_payment($order_id): array
+    {
+        $order = \wc_get_order($order_id);
+
+        $order->update_status('processing', __('Afventer færdiggørelse', 'mt-wc-economic'));
+
+        WC()->cart->empty_cart();
+
+        return [
+            'result' => 'success',
+            'redirect' => $this->get_return_url($order),
+        ];
+    }
+
+    private function createEconomicInvoice(int $orderId): void
+    {
+
+        $order = wc_get_order($orderId);
+
+        $paymentMethod = $order->get_payment_method();
+
+        if ($paymentMethod !== 'economic_invoice') {
+            return;
+        }
+
+        ActionScheduleService::addCreateInvoiceJob($order);
+    }
+
+    public function onNewOrder(int $orderId): void
+    {
+        if ($this->get_option('economic_invoice_event') === 'creation') {
+            $this->createEconomicInvoice($orderId);
+        }
+    }
+
+    public function onOrderCompleted(int $orderId): void
+    {
+        if ($this->get_option('economic_invoice_event') === 'completed') {
+
+            $this->createEconomicInvoice($orderId);
+        }
+    }
+
+    private function getLayouts(): array
+    {
+        return Layout::all()->mapWithKeys(function ($layout) {
```

Comment:
Burde måske ligge i en service for sig selv. og inkluderer error handling.

Gælder også de andre economic getters.